framework/web_app/src/pages/StockOverview.vue

```diff
@@ -1,96 +1,260 @@
+                    :items="stockLevelStore.stockLevels"
+                    dense
+                    dropdown-icon="none"
+                    no-caps
+                    push
+                    rounded
+                    style="width: 28px"
+                >
+                    <q-item
+                        :key="level.sequence"
+                        @click="
+                            updateStockLevelAsync(
+                                item.stock_item_id,
+                                level.stock_level_id
+                            )
+                        "
+                        clickable
+                        v-close-popup
+                        v-for="level in stockLevelStore.stockLevels"
+                    >
+                        <q-item-section avatar>
+                            <q-avatar
+                                :color="
+                                    getStockLevelColour(level.stock_level_id)
+                                "
+                                size="25px"
+                            />
+                        </q-item-section>
+
+                        <q-item-section>
+                            <q-item-label>{{ level.description }}</q-item-label>
+                        </q-item-section>
+                    </q-item>
+                </q-btn-dropdown>
+                <q-btn
+                    class="q-mx-sm"
+                    @click="shouldDisplayAddToShoppingCartModal = true"
+                    flat
+                    icon="shopping_cart"
+                    rounded
+                />
+
+                <q-separator
+                    class="q-ml-sm q-mr-md"
+                    vertical
+                />
+                <p class="text-bold q-ma-sm">{{ item.name }}</p>
+            </q-card-actions>
+
+            <q-card-section class="col">
+                <p class="text-weight-bold q-ma-none">Location</p>
+                <p class="q-ma-none">{{ 'item.stock_location' }}</p>
+            </q-card-section>
+
+            <q-card-section class="col">
+                <p class="text-weight-bold q-ma-none">Stock Group</p>
+                <p class="q-ma-none">{{ 'item.stock_group' }}</p>
+            </q-card-section>
+        </q-card-section>
+    </q-card>
+
+    <q-dialog v-model="shouldDisplayCreateStockItemModal">
+        <q-card
+            class="q-pa-md"
+            style="width: 700px; max-width: 80vw"
+        >
+            <p class="text-h4">Add a new stock item</p>
+            <q-form
+                class="q-gutter-md"
+                @submit="
+                    createStockItemAsync(createStockItemForm).then(
+                        clearCreateStockItemForm
+                    )
+                "
+            >
+                <q-input
+                    :rules="[
+                        (val: string) =>
+                            (val && val.length > 0) || 'Please type something'
```

Comment:
Wasn't sure if this would also be these inline functions would also be un-performant but GPT seems to think so. Refer to other inline func comment for context if you saw this first.

_'Each time the component re-renders, the inline function within the array is redefined. This results in a new function object being created on every render.'_

In general I try to avoid having logic in the UI for debugging & UI/logic separation but that is more preference than a must.

Reply:
The rules have been moved out.

Also move the simple  A = true logic out of the UI into event handlers. For a simple scenario like this it is not a matter of performance but a matter separation of concerns between UI Components/Elements/Controls and their logic